content/docs/user-guide/interactivity/robotics/importing-robot/sdformat-sensors.md

```diff
@@ -0,0 +1,66 @@
+| _GNSSSensorHook_   | `gps`, `navsat`            | `libgazebo_ros_gps_sensor`    | `ROS2GNSSSensor`            |
+| _ImuSensorHook_    | `imu`                      | `libgazebo_ros_imu_sensor`    | `ROS2ImuSensorComponent`    |
+| _LidarSensorHook_  | `lidar`, `ray`             | `libgazebo_ros_ray_sensor`    | `ROS2LidarSensorComponent`  |
+|                    | `gpu_lidar`, `gpu_ray`     | `libgazebo_ros_laser`         | _see information below_     |
+
+GPU implementation of ray/lidar sensors is not included in ROS 2 Gem and requires additional [O3DE RGL Gem](https://github.com/RobotecAI/o3de-rgl-gem). GPU sensors are mapped during the import to run on a CPU if this Gem is not available.
+
+### Extending default mapping
+
+You can extend the default mapping by implementing additional _hooks_ and registering them in the system based on the _SerializeContext_ reflection system. Typically, this consists of three tasks.
+
+First, you need to declare `ROS2::SDFormat::SensorImporterHook` structure that consists of the following:
+* set of SDFormat sensors associated with your import scheme
+* set of plugin names associated with your import scheme
+* set of the supported parameters in the input robot description (used for import verbose only)
+* registered callback function that is invoked by Robot Importer when _hook's_ definition matches the input data
+
+The registered callback function creates a number of O3DE components that are necessary to simulate a particular sensor. Additionally, it parses the robot description file to read certain processing parameters. 
+
+Your second task is to implement your desired simulation behavior in an O3DE component, which would be created by the Robot Importer. Finally, you need to define and register your _hook_ via the _SerializeContext_ reflection system using `SensorImporterHooks` attribute tag. This allows the Robot Importer to find and add your _hook_ to the mapping. The registration can be done in any O3DE editor component and multiple _hooks_ can be registered at once. For simplicity, you might want to add the registration directly to your O3DE component. A sample code implementing the registration scheme can be as follows:
+```cpp
+void YourO3DEEditorComponent::Reflect(AZ::ReflectContext* context)
+{
+    if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+    {
+        const ROS2::SDFormat::SensorImporterHook& yourImporterHook = YourMethodToCreateHook();
+        const ROS2::SDFormat::SensorImporterHook& anotherImporterHook = YourAnotherMethodToCreateHook();
+        serializeContext->Class<YourO3DEEditorComponent, BaseClassOfYourO3DEEditorComponent>()
+                ->Attribute(
+                    "SensorImporterHooks",
+                    ROS2::SDFormat::SensorImporterHooksStorage{ AZStd::move(yourImporterHook), AZStd::move(anotherImporterHook) });
+    }
+    // more reflection code goes here
+}
+```
+
```

Comment:
Perhaps add a link to RGL Gem code where the hook is implemented as an example.

Reply:
There is no separate hook for RGL Gem; the same hook is used for GPU- and CPU-based implementations of LiDAR.